tools/hangar/__snapshots__/test_corpus/valid/inflight_class_outside_inflight_closure.test.w_compile_tf-aws.md

```diff
@@ -10,7 +10,7 @@ module.exports = function({ $BinaryOperation }) {
       return $obj;
     }
     async handle() {
-      const op = new $BinaryOperation(10,20);
+      const op = (await (async () => {const o = new $BinaryOperation(); if ('$inflight_init' in o) { await o.$inflight_init(10,20); } return o; })());
```

Comment:
It might be possible to dress this up a little nicer by defining a helper method that is responsible for both `new`-ing the class and then performing the inflight initialization logic:

```suggestion
      const op = (await $BinaryOperation.$init(10, 20));
```

(basically, I think this means changing `$inflight_init` to be a static method. What do you think?)

Reply:
Implementing this is tricky because I'll need to detect references to `this` in the inflight init and special case them.
Having some internal stdlib function for `std.$instantial_inflight_type(Type, args)` that calls `new`, optionally calls `$inflight_init` and finally returns the object seems ideal to me. But this is yet another case of needing access to stdlib from inflight which we currently don't have..

Reply:
I cleaned this up a bit using `?.` optional chaining. I think it's good enough for now given the challenges above. WDYT?

Reply:
> Implementing this is tricky because I'll need to detect references to this in the inflight init and special case them.

Ah, I see what you mean - if a class has code like this:

```js
inflight class A {
  init() {
    this.x = this.calculate();
  }
  calculate() {
    return 42;
  }
}
```

Then the JS code would have to be something like:

```js
class A {
  constructor() {}
  static $init(..args) {
    const $this = new A(..args);
    $this.x = $this.calculate();
  }
}
```

(`$this` could be also named `o`, or `self`, or whatever)

Doable but it's not trivial. I also agree `std.$createInflight(Type, args)` would also be a good solution to this. Yeah I'm cool with revisiting this once we have easy access to stdlib inflight 👍